src/analyses/apron/apronAnalysis.apron.ml

```diff
@@ -418,6 +418,14 @@ struct
         | Some lv -> invalidate_one ask ctx st' lv
         | None -> st'
       )
+    | ThreadExit _, _ ->
+      begin match ThreadId.get_current ask with
+        | `Lifted tid ->
+          ignore @@ Priv.thread_return ask ctx.global ctx.sideg tid st;
```

Comment:
Looking at just this, it seems something `return`-like should be happening here, i.e. to remove locals from `st`. The only reason it's fine right now is that no `Priv.thread_return` uses `st.apr`. Would be worth adding a comment about it here.

Reply:
Good point, I added a comment!